src/datetime.rs

```diff
@@ -356,15 +356,16 @@ impl DateTime {
                         // fraction is either:
                         // - up to 3 digits of millisecond fractions, i.e. microseconds
                         // - or up to 6 digits of second fractions, i.e. milliseconds
-                        let max_digits = if timestamp > MS_WATERSHED { 3 } else { 6 };
+                        let ts_abs = timestamp.checked_abs().ok_or(ParseError::DateTooSmall)?;
```

Comment:
I feel like it's ok to saturate here, because we only use it for scale.

```suggestion
                        let ts_abs = timestamp.saturating_abs();
```